netbox/extras/signals.py

```diff
@@ -70,13 +70,21 @@ def handle_changed_object(sender, instance, **kwargs):
 
     # Record an ObjectChange if applicable
     if m2m_changed:
-        ObjectChange.objects.filter(
+        qs = ObjectChange.objects.filter(
             changed_object_type=ContentType.objects.get_for_model(instance),
             changed_object_id=instance.pk,
             request_id=request.id
-        ).update(
-            postchange_data=instance.to_objectchange(action).postchange_data
         )
+        if not qs:
+            objectchange = instance.to_objectchange(action)
+            if objectchange and objectchange.has_changes:
+                objectchange.user = request.user
+                objectchange.request_id = request.id
+                objectchange.save()
+        else:
+            qs.update(
+                postchange_data=instance.to_objectchange(action).postchange_data
+            )
```

Comment:
We can clean this up a bit using the [`update_or_create()`](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#update-or-create) queryset method to automatically create a new change record where needed. We can also pull `objectchange = instance.to_objectchange(action)` up to the top, as it appears to be needed under every condition.

Reply:
I moved the objectchange = line.

For update_or_create I don't think that would be any cleaner as on create you are doing objectchange.save and not specifying all the fields.  For update_or_create you would need to specify all the fields for defaults and create_defaults which is sort of duplicating instance.to_objectchange.